daemons/gptp/common/ieee1588port.cpp

```diff
@@ -79,47 +82,73 @@ IEEE1588Port::~IEEE1588Port()
 	if( qualified_announce != NULL ) delete qualified_announce;
 }
 
-IEEE1588Port::IEEE1588Port
-(IEEE1588Clock * clock, uint16_t index, bool forceSlave,
- int accelerated_sync_count, HWTimestamper * timestamper, int32_t offset,
- InterfaceLabel * net_label, OSConditionFactory * condition_factory,
- OSThreadFactory * thread_factory, OSTimerFactory * timer_factory,
- OSLockFactory * lock_factory)
+IEEE1588Port::IEEE1588Port(IEEE1588PortInit_t *portInit)
 {
 	sync_sequence_id = 0;
 
-	clock->registerPort(this, index);
-	this->clock = clock;
-	ifindex = index;
+	clock = portInit->clock;
+	ifindex = portInit->index;
+	clock->registerPort(this, ifindex);
 
-	this->forceSlave = forceSlave;
+	forceSlave = portInit->forceSlave;
 	port_state = PTP_INITIALIZING;
 
-	asCapable = false;
+	automotive_profile = portInit->automotive_profile;
+	isGM = portInit->isGM;
+	testMode = portInit->testMode;
+	initialLogSyncInterval = portInit->initialLogSyncInterval;
+	initialLogPdelayReqInterval = portInit->initialLogPdelayReqInterval;
+	operLogPdelayReqInterval = portInit->operLogPdelayReqInterval;
+	operLogSyncInterval = portInit->operLogSyncInterval;
+
+	if (automotive_profile) {
+		asCapable = true;
+
+		if (initialLogSyncInterval == LOG2_INTERVAL_INVALID)
+			initialLogSyncInterval = -5;     // 31.25 ms
+		if (initialLogPdelayReqInterval == LOG2_INTERVAL_INVALID)
+			initialLogPdelayReqInterval = -3;  // 125 ms
+		if (operLogPdelayReqInterval == LOG2_INTERVAL_INVALID)
+			operLogPdelayReqInterval = 0;      // 1 second
+		if (operLogSyncInterval == LOG2_INTERVAL_INVALID)
+			operLogSyncInterval = 0;           // 1 second
+	}
+	else {
+		asCapable = false;
+
+		if (initialLogSyncInterval == LOG2_INTERVAL_INVALID)
+			initialLogSyncInterval = -3;       // 125 ms
```

Comment:
I might be missing something here, but dont we have a constant somewhere for logsyncinterval (and the others as well) ? Then we wouldnt have to hard code these values here (-5, -3 etc).